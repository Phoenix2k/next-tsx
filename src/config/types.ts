export interface Config {
  a11yEnv: Array<string>;
  pages: Pages;
  theme: Theme;
}

export interface Page {
  navLabel: string;
  slug: string;
  title: string;
  url: string;
}

export interface Pages {
  [key: string]: Page;
}

export interface Theme {
  classes: {
    navLinkActive: string;
  };
  colors: {
    backgroundColor: string;
    borderColor: string;
    color: string;
    navLink: string;
    navLinkActive: string;
  };
  typography: {
    sansSerif: string;
  };
}
