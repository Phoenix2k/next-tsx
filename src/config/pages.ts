import sitemap from '../../data/sitemap.json';
import { Pages } from './types';

const pages: Pages = sitemap;

export default pages;
