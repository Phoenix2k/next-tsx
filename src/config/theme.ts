import { Theme } from './types';

const theme: Theme = {
  classes: {
    navLinkActive: 'nav-link--active'
  },
  colors: {
    backgroundColor: 'rgb(255,255,255)',
    borderColor: '#ccc',
    color: 'black',
    navLink: 'blue',
    navLinkActive: 'rgb(0, 128, 0)'
  },
  typography: {
    sansSerif: '"Open Sans", Helvetica, Arial, sans-serif'
  }
};

export default theme;
