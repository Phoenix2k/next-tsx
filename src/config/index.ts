import pages from './pages';
import theme from './theme';
import { Config } from './types';

const config: Config = {
  a11yEnv: ['development'],
  pages,
  theme
};

export default config;
