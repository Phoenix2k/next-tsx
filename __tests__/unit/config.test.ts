/// <reference types="jest"/>

import config from '../../src/config';

describe('Config', () => {
  it('matches specification', () => {
    expect(Object.keys(config).length).toEqual(3);
    expect(Object.keys(config.pages).length).toBeGreaterThan(0)
    expect(Object.keys(config.theme).length).toBeGreaterThan(0)
  });
  it('matches snapshot', () => {
    expect(config).toMatchSnapshot();
  });
});
