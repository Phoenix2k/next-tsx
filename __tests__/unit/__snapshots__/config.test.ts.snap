// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config matches snapshot 1`] = `
Object {
  "a11yEnv": Array [
    "development",
  ],
  "pages": Object {
    "about": Object {
      "navLabel": "About",
      "slug": "about-page",
      "title": "About page",
      "url": "/about",
    },
    "home": Object {
      "navLabel": "Home",
      "slug": "home-page",
      "title": "Home page",
      "url": "/",
    },
  },
  "theme": Object {
    "classes": Object {
      "navLinkActive": "nav-link--active",
    },
    "colors": Object {
      "backgroundColor": "rgb(255,255,255)",
      "borderColor": "#ccc",
      "color": "black",
      "navLink": "blue",
      "navLinkActive": "rgb(0, 128, 0)",
    },
    "typography": Object {
      "sansSerif": "\\"Open Sans\\", Helvetica, Arial, sans-serif",
    },
  },
}
`;
